pkg/isoeditor/kargs_test.go

```diff
@@ -0,0 +1,133 @@
+package isoeditor
+
+import (
+	"errors"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	kargsConfileFile = `
+{
+  "default": "mitigations=auto,nosmt coreos.liveiso=fedora-coreos-35.20220103.3.0 ignition.firstboot ignition.platform.id=metal",
+  "files": [
+    {
+      "offset": 970,
+      "path": "EFI/fedora/grub.cfg"
+    },
+    {
+      "offset": 1870,
+      "path": "isolinux/isolinux.cfg"
+    }
+  ],
+  "size": 1137
+}
+`
+	grubFileWithEmbedArea = `
+function load_video {
+  insmod efi_gop
+}
+
+insmod ext2
+
+set timeout=5
+### END /etc/grub.d/00_header ###
+
+### BEGIN /etc/grub.d/10_linux ###
+menuentry 'Fedora CoreOS (Live)' --class fedora --class gnu-linux --class gnu --class os {
+	linux /images/pxeboot/vmlinuz mitigations=auto,nosmt coreos.liveiso=fedora-coreos-35.20220103.3.0 ignition.firstboot ignition.platform.id=metal
+################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################ COREOS_KARG_EMBED_AREA
+	initrd /images/pxeboot/initrd.img /images/ignition.img
+}
+`
+	grubFileWithoutEmbedArea = `
+function load_video {
+  insmod efi_gop
+}
+
+insmod ext2
+
+set timeout=5
+### END /etc/grub.d/00_header ###
+
+### BEGIN /etc/grub.d/10_linux ###
+menuentry 'Fedora CoreOS (Live)' --class fedora --class gnu-linux --class gnu --class os {
+	linux /images/pxeboot/vmlinuz mitigations=auto,nosmt coreos.liveiso=fedora-coreos-35.20220103.3.0 ignition.firstboot ignition.platform.id=metal
+	initrd /images/pxeboot/initrd.img /images/ignition.img
+}
+`
+)
+
+var _ = Describe("kargs tests", func() {
```

Comment:
I think these should be separate `Describe` blocks.

Each `Describe` should list a particular function. So here I'd expect something like:

```go
var _ = Context("with karg helpers", func() {
  const(
    ...
  )

  mockFileReaderCreator := func...
  // more helpers

  Describe("kargsFiles", func() {
    ...
  })

  Describe("kargsEmbedAreaBoundariesFinder", func() {
    ...
  })
})
```

That also gets the constants out of the package scope so they can't accidentally be referenced by non-test code. It also helps with some auto complete engines that might be able to read the scope of identifiers or something.

Reply:
BTW, go does not allow reference from non-test code to test code.
Will change.

Reply:
Switched `Describe` and `Context`